src/signature/mod.rs

```diff
@@ -24,6 +29,51 @@ impl TryFrom<u8> for SignatureVersion {
     }
 }
 
+/// copied from wasm/wasm-crypto
+fn create_from_pair(pair: &[u8]) -> EdKeypair {
+    match EdKeypair::from_bytes(pair) {
+        Ok(pair) => return pair,
+        Err(_) => panic!("Provided pair is invalid."),
+    }
+}
+
+/// copied from wasm/wasm-crypto
+fn create_from_parts(pubkey: &[u8], seckey: &[u8]) -> EdKeypair {
+    let mut pair = vec![];
+
+    pair.extend_from_slice(seckey);
+    pair.extend_from_slice(pubkey);
+
+    create_from_pair(&pair)
+}
+
+/// copied from wasm/wasm-crypto
+fn create_public(pubkey: &[u8]) -> Ed25519Pub {
+    match Ed25519Pub::from_bytes(pubkey) {
+        Ok(pubkey) => return pubkey,
+        Err(_) => panic!("Provided public key is invalid."),
+    }
+}
+
+/// copied from wasm/wasm-crypto
+fn create_from_seed(seed: &[u8]) -> EdKeypair {
+    let seckey = EdSecretKey::from_bytes(seed).unwrap();
+    let pubkey: Ed25519Pub = (&seckey).into();
+
+    create_from_parts(pubkey.as_bytes(), seed)
+}
+
+/// copied from wasm/wasm-crypto
+pub fn create_ed_keypair_from_seed(seed: &[u8]) -> Vec<u8> {
+    create_from_seed(seed).to_bytes().to_vec()
+}
+
+/// Sign an ed25519 signature
+pub fn ed_sign(keypair: EdKeypair, payload: &[u8]) -> Vec<u8> {
```

Comment:
function signature should be: `sign_ed25519(<private key>: &[u8], <payload>: &[u8]) -> Vec<u8>`
This will give the caller the most flexibility to provide the private key however they want. Otherwise it's unnecessarily restricted to use `EdKeypair` crate / struct only.

next step this function should turn into a method on doughnut structs.
to use like: `doughnut.sign_ed25519(<private_key>)`
A trait will help, then impl the trait for the Doughnut structs to sign and append the signature
e.g.
```
/// Doughnut signing methods
trait Signing {
    /// sign using Sr25519 method
    fn sign_sr25519(&mut self, private_key: &[u8]);
    /// sign using Ed25519 method
    fn sign_ed25519(&mut self, private_key: &[u8]);
}
```